components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAOTest.java

```diff
@@ -0,0 +1,1486 @@
+                "Country", "", true);
+        Claim remoteClaim = new Claim();
+        remoteClaim.setClaimId(0);
+        remoteClaim.setClaimUri("Country");
+        claimConfig.setClaimMappings(new ClaimMapping[]{claimMapping});
+        claimConfig.setIdpClaims(new Claim[]{remoteClaim});
+        idp1.setClaimConfig(claimConfig);
+
+        // Initialize Test Identity Provider 2.
+        IdentityProvider idp2 = new IdentityProvider();
+        idp2.setIdentityProviderName("testIdP2");
+        idp2.setHomeRealmId("2");
+
+        ClaimConfig claimConfig2 = new ClaimConfig();
+        claimConfig2.setLocalClaimDialect(true);
+        claimConfig2.setRoleClaimURI("http://wso2.org/claims/role");
+        claimConfig2.setUserClaimURI("http://wso2.org/claims/fullname");
+        ClaimMapping claimMapping2 = new ClaimMapping();
+        Claim localClaim2 = new Claim();
+        localClaim2.setClaimId(0);
+        localClaim2.setClaimUri("http://wso2.org/claims/fullname");
+        claimMapping2.setLocalClaim(localClaim2);
+        claimConfig2.setClaimMappings(new ClaimMapping[]{claimMapping2});
+        idp2.setClaimConfig(claimConfig2);
+
+        // Initialize Test Identity Provider 3.
+        IdentityProvider idp3 = new IdentityProvider();
+        idp3.setIdentityProviderName("testIdP3");
+        idp3.setHomeRealmId("3");
+
+        // IDP with PermissionsAndRoleConfig, FederatedAuthenticatorConfigs, ProvisioningConnectorConfigs, ClaimConfigs.
+        idPManagementDAO.addIdP(idp1, SAMPLE_TENANT_ID);
+        // IDP with Local Cliam Dialect ClaimConfigs.
+        idPManagementDAO.addIdP(idp2, SAMPLE_TENANT_ID);
+        // IDP with Only name.
+        idPManagementDAO.addIdP(idp3, SAMPLE_TENANT_ID2);
+    }
+
+    private int getIdPCount(Connection connection, String idpName, int tenantId) throws SQLException {
+
+        String query = IdPManagementConstants.SQLQueries.GET_IDP_BY_NAME_SQL;
+        PreparedStatement statement = connection.prepareStatement(query);
+        statement.setInt(1, tenantId);
+        statement.setInt(2, MultitenantConstants.SUPER_TENANT_ID);
+        statement.setString(3, (idpName));
+        ResultSet resultSet = statement.executeQuery();
+        int resultSize = 0;
+        if (resultSet.next()) {
+            resultSize = resultSet.getRow();
+        }
+        statement.clearParameters();
+        statement.close();
+        return resultSize;
+    }
+
+    private IdentityProvider idpFromCacheByName(String idpName) {
+        IdPCacheByName idPCacheByName = IdPCacheByName.getInstance();
+        IdPNameCacheKey cacheKey = new IdPNameCacheKey(idpName);
+        IdPCacheEntry entry = idPCacheByName.getValueFromCache(cacheKey, TENANT_DOMAIN);
+        if (entry != null) {
+            return entry.getIdentityProvider();
+        } else {
+            return null;
+        }
+    }
+
+    private IdentityProvider idpFromCacheByResourceId(String resourceId) {
```

Comment:
Add new line after the method. Please check all places

Reply:
Resolved with https://github.com/wso2/carbon-identity-framework/pull/4110/commits/28a1194ed9f5de659bd1879f625cd070efdac3fa

Reply:
Resolved with https://github.com/wso2/carbon-identity-framework/pull/4110/commits/28a1194ed9f5de659bd1879f625cd070efdac3fa